server.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/golang/glog"
+	"k8s.io/api/admission/v1beta1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"github.com/casbin/casbin/v2"
+)
+
+type CasbinServerHandler struct {
+}
+
+func (gs *CasbinServerHandler) serve(w http.ResponseWriter, r *http.Request) {
+	var body []byte
+	if r.Body != nil {
+		if data, err := ioutil.ReadAll(r.Body); err == nil {
+			body = data
+		}
+	}
+	if len(body) == 0 {
+		glog.Error("empty body")
+		http.Error(w, "empty body", http.StatusBadRequest)
+		return
+	}
+	glog.Info("Received request")
+
+	if r.URL.Path != "/validate" {
+		glog.Error("no validate")
+		http.Error(w, "no validate", http.StatusBadRequest)
+		return
+	}
+
+	arRequest := v1beta1.AdmissionReview{}
+	if err := json.Unmarshal(body, &arRequest); err != nil {
+		glog.Error("incorrect body")
+		http.Error(w, "incorrect body", http.StatusBadRequest)
+	}
+
+	raw := arRequest.Request.Object.Raw
+	pod := v1.Pod{}
+	if err := json.Unmarshal(raw, &pod); err != nil {
+		glog.Error("error deserializing pod")
+		return
+	}
+	allowed, err := e.Enforce(request)
```

Comment:
Here's the actual logic part, which would validate the users based on thei roles.

Reply:
How is this code controling the access to k8s?

Reply:
Yes, this would be deployed inside the clusters. This could be done by applying the [deployment.yaml](https://github.com/casbin/k8s-authz/blob/b5e67d7e91d46bbc105dc81a2441b2734a8126fc/manifests/deployment.yaml) through kubectl. 

`kubectl apply -f deployment.yaml`